File: GP Analysis/update-data.js
#!/usr/bin/env node
/**
 * Lee config.json (csvPath) y genera data.js con el contenido del CSV.
 * Uso: node update-data.js
 * Para cambiar el CSV: editá config.json (csvPath) y volvé a ejecutar este script.
 */

const fs = require('fs');
const path = require('path');

const dir = __dirname;
const configPath = path.join(dir, 'config.json');
const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
const csvPath = path.resolve(dir, config.csvPath);

if (!fs.existsSync(csvPath)) {
  console.error('No se encontró el CSV en:', csvPath);
  console.error('Revisá config.json → csvPath.');
  process.exit(1);
}

const csv = fs.readFileSync(csvPath, 'utf8');
// Escapar para string en JS: \ -> \\, ` -> \`, $ -> \$, fin de línea y comillas
const escaped = csv
  .replace(/\\/g, '\\\\')
  .replace(/`/g, '\\`')
  .replace(/\$/g, '\\$')
  .replace(/\r\n/g, '\n')
  .replace(/\r/g, '\n');

const out = `// Generado por update-data.js desde config.json → csvPath. No editar a mano.
window.TALENT_POOL_CSV = \`${escaped}\`;
`;

fs.writeFileSync(path.join(dir, 'data.js'), out, 'utf8');
console.log('OK: data.js generado desde', config.csvPath);

File: GP Analysis/update_data.py
#!/usr/bin/env python3
"""
Lee config.json (csvPath) y genera data.js con el contenido del CSV.
Uso: python3 update_data.py
Para cambiar el CSV: editá config.json (csvPath) y volvé a ejecutar este script.
"""
import json
import os

dir_path = os.path.dirname(os.path.abspath(__file__))
config_path = os.path.join(dir_path, "config.json")
with open(config_path, encoding="utf-8") as f:
    config = json.load(f)
csv_path = os.path.normpath(os.path.join(dir_path, config["csvPath"]))

if not os.path.isfile(csv_path):
    print("No se encontró el CSV en:", csv_path)
    print("Revisá config.json → csvPath.")
    raise SystemExit(1)

with open(csv_path, encoding="utf-8") as f:
    csv_content = f.read()
# Escapar para template literal JS: \ ` $
escaped = csv_content.replace("\\", "\\\\").replace("`", "\\`").replace("$", "\\$")
escaped = escaped.replace("\r\n", "\n").replace("\r", "\n")

out = """// Generado por update_data.py (o update-data.js) desde config.json → csvPath. No editar a mano.
window.TALENT_POOL_CSV = `""" + escaped + "`;\n"

data_js_path = os.path.join(dir_path, "data.js")
with open(data_js_path, "w", encoding="utf-8") as f:
    f.write(out)
print("OK: data.js generado desde", config["csvPath"])
